src/main/java/org/jcodec/containers/mp4/muxer/CodecMP4MuxerTrack.java

```diff
@@ -153,9 +151,15 @@ void addVideoSampleEntry(VideoCodecMeta meta) {
 
     public void setCodecPrivateIfNeeded() {
         if (codec == Codec.H264) {
-            getEntries().get(0).add(H264Utils.createAvcCFromPS(selectUnique(spsList), selectUnique(ppsList), 4));
+            List<ByteBuffer> sps = selectUnique(spsList);
+            List<ByteBuffer> pps = selectUnique(ppsList);
+            if (!sps.isEmpty() && !pps.isEmpty()) {
+                getEntries().get(0).add(H264Utils.createAvcCFromPS(sps, pps, 4));
+            }
```

Comment:
I would also add a warning here:
} else {
  Logger.warn("Not adding a sample entry for h.264 track, missing any SPS/PPS NAL units");
}